draft-rfcs/const-evaluatable-checked.md

```diff
@@ -0,0 +1,12 @@
+# const evaluatable checked
+
+TODO: All of this :)
+
+## Requirements
+
+For this to work, const operations have to be both pure and deterministic, at least when they are used in the type system.
+
+FIXME: const allocations?
+FIXME: non-det floats, are they are problem? We use `ap_float` which is hopefully deterministic.
```

Comment:
Floats in the CTFE/Miri engine are completely deterministic. However, they can sometimes produce different results than what would happen on the real hardware, in particular when NaNs are involved. See https://github.com/rust-lang/unsafe-code-guidelines/issues/237 for a collection of issues around floating-point semantics.